.github/workflows/cd.yml

```
@@ -7,12 +7,40 @@ on:
+        image: redis:latest
+        ports:
+          - 6379:6379
+
     steps:
       - uses: actions/checkout@v4
         with:
           token: ${{ secrets.HS_TOKEN }}
           submodules: recursive
 
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
```

actionlint (1.7.7)

the runner of "actions/setup-java@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)